tests/partial_model_tests.rs

```diff
@@ -66,3 +69,51 @@ struct NestOption {
     #[sea_orm(nested)]
     _foo: Option<SimpleTest>,
 }
+
+#[sea_orm_macros::test]
+async fn partial_model_left_join_does_not_exist() {
+    use common::bakery_chain::*;
+
+    #[derive(FromQueryResult, DerivePartialModel)]
+    #[sea_orm(entity = "bakery::Entity")]
+    struct Bakery {
+        id: i32,
```

Comment:
The issue occurs because this `id` field (and the `name` field below) have the same name as the ones in the `Cake` struct. SQLx does not differentiate between these two and just overrides the ones it encounters first (the ones for `Cake`) with the ones in it encounters next (the ones for `Bakery`). The "good" case (where the associated `Bakery` exists) is arguably the most broken, as the `id` and `name` values of the `Cake` get silently replaced with the ones of the `Bakery`.